docs/proposals/34-observability2.0-opentelemtry-operator-and-collectors.md

```diff
@@ -0,0 +1,156 @@
+
+---
+
+## Proposal
+
+This section outlines the implementation plan to introduce the OpenTelemetry operator and collectors into Gardener's shoot control plane namespaces and in shoot nodes as a systemd service, aligning with Step 2 of the *Observability 2.0* roadmap.
+
+### 1. Deploy the OpenTelemetry Operator on Seed Clusters
+
+- **Objective:** Establish centralized management of OpenTelemetry collectors across all shoot control planes.
+- **Details:** Deploy the OpenTelemetry operator (e.g., from the [OpenTelemetry Operator for Kubernetes](https://github.com/open-telemetry/opentelemetry-operator)) in the `garden` namespace of each seed cluster using `ManagedResources`, similar to the Prometheus operator deployment in *GEP-19*.
+- **Configuration:** The operator will be deployed following the `botanist` workflow in the same way as the current `prometheus` and `fluent-bit` operators. This meaning that during the `Seed` reconciliation flow, the OpenTelemetryOperator will be deployed into the `garden` namespace.
+
+### 2. Create OpenTelemetry Collector Instances per Shoot
+
+- **Objective:** Provide each shoot control plane with a dedicated collector to process its observability data.
+- **Details:** For each shoot, deploy an OpenTelemetry collector instance in its control plane namespace (e.g., `shoot--project--name`). The collector will run as a `Deployment` managed by the OpenTelemetry operator.
+- Configuration: During the `Shoot` reconciliation flow, an [`OpenTelemetryCollector` Custom Resource](https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1beta1/opentelemetrycollector_types.go) will be created in the `Shoot` namespace. This Custom Resource will trigger the Operator to create an OpenTelemetry Collector Deployment & Ingress inside the namespace.
+- **Example OpenTelemetryCollector CR:**
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  labels:
+    resources.gardener.cloud/managed-by: gardener
+  name: opentelemetry-collector
+spec:
+  mode: deployment
+  config:
+    exporters:
+      loki:
+      endpoint: https://v-local--local.ingress.local.seed.local.gardener.cloud/vali/api/v1/push
+        headers:
+        # Authenticate against Vali using a Bearer Token
+        Authorization: "Bearer ${file:{{ .pathAuthToken }}}"
+    receivers:
+      otlp:
+        protocols:
+          http:
+            endpoint: 0.0.0.0:4317
+    service:
+      pipelines:
+        logs:
+          exporters:
+          - loki
+          receivers:
+          - otlp
+```
+
+![Figure 1](assets/gep-34-otel-operator-architecture.png)
+
+### 3. Integrate with Existing Components
+
+- **Objective:** Ensure a smooth transition by interfacing with the current observability stack.
+- **Details:**
+  - **Fluent-bit Integration:** Configure Fluent-bit to forward logs to the collector’s `loki` receiver, maintaining compatibility with the Vali-based setup from *GEP-19*.
```

Comment:
Please add a link.

Reply:
We usually link between GEPs in case they refer to each other. Hence, please add a link to GEP-19.